tests/phpunit/tests/test-create-delete-languages.php

```diff
@@ -263,8 +324,114 @@ public function test_create_language_object_without_term_language_tax() {
+		$this->pll_env->terms = new PLL_CRUD_Terms( $this->pll_env );
+
+		$args = array(
+			'name'       => 'English',
+			'slug'       => 'en',
+			'locale'     => 'en_US',
+			'rtl'        => 0,
+			'flag'       => 'us',
+			'term_group' => 0,
+		);
+		$this->assertTrue( $this->pll_env->model->add_language( $args ) );
+
+		$args = array(
+			'name'       => 'Français',
+			'slug'       => 'fr',
+			'locale'     => 'fr_FR',
+			'rtl'        => 0,
+			'flag'       => 'fr',
+			'term_group' => 1,
+		);
+		$this->assertTrue( $this->pll_env->model->add_language( $args ) );
+		$this->assertEquals( 'en', $this->pll_env->options['default_lang'] );
+
+		$fr = self::factory()->post->create();
+		$this->pll_env->model->post->set_language( $fr, 'fr' );
+
+		$lang = $this->pll_env->model->get_language( 'fr' );
+		$this->pll_env->model->delete_language( $lang->term_id );
+		$this->assertCount( 1, $this->pll_env->model->get_languages_list() );
+	}
+
+	/**
+	 * Test a second language deletion with 'term_group' > 0
+	 * and the language is assigned to a content.
+	 *
+	 * A language cache clean up and a languages list built are also run during the language deletion process.
+	 *
+	 * Polylang Pro #1626
+	 */
+	public function test_delete_language_with_content_which_has_this_language_and_with_clean_languages_cache() {
+		add_action(
+			'pre_delete_term',
+			function( $term_id ) {
+				$this->pll_env->model->clean_languages_cache();
+				$this->pll_env->model->get_languages_list();
+			}
+		);
+
+		// first language
+		$args = array(
+			'name'       => 'English',
+			'slug'       => 'en',
+			'locale'     => 'en_US',
+			'rtl'        => 0,
+			'flag'       => 'us',
+			'term_group' => 0,
+		);
+		$this->assertTrue( $this->pll_env->model->add_language( $args ) );
+
+		$args = array(
+			'name'       => 'Español',
+			'slug'       => 'es',
+			'locale'     => 'es_ES',
+			'rtl'        => 0,
+			'flag'       => 'es',
+			'term_group' => 3,
+		);
+		$this->assertTrue( $this->pll_env->model->add_language( $args ) );
+
+		$args = array(
+			'name'       => 'Deutsch',
+			'slug'       => 'de',
+			'locale'     => 'de_DE',
+			'rtl'        => 0,
+			'flag'       => 'de',
+			'term_group' => 2,
+		);
+		$this->assertTrue( $this->pll_env->model->add_language( $args ) );
+
+		$args = array(
+			'name'       => 'Français',
+			'slug'       => 'fr',
+			'locale'     => 'fr_FR',
+			'rtl'        => 0,
+			'flag'       => 'fr',
+			'term_group' => 1,
+		);
+		$this->assertTrue( $this->pll_env->model->add_language( $args ) );
+		$this->assertEquals( 'en', $this->pll_env->options['default_lang'] );
+
+		$fr = self::factory()->post->create();
+		$this->pll_env->model->post->set_language( $fr, 'fr' );
+
+		$lang = $this->pll_env->model->get_language( 'fr' );
+		$this->pll_env->model->delete_language( $lang->term_id );
+		$this->assertCount( 3, $this->pll_env->model->get_languages_list() );
+		remove_all_actions( 'pre_delete_term' );
```

Comment:
This is useless. Your hook is remove in `tear_down()` by WP.
```suggestion
```

Reply:
No!
In fact, action declared in this test is run when we delete the all languages in the `tear_down()` method which triggers a notice on `options['default_lang']` when we delete the last language because this option doesn't exist anymore.

Move clean up of this action in the `tear_down()` method and so replace the closure with a new method.

Done in 11b4be70ba5afc07d35339a42df5fd51c83e2d9a